packages/swap/src/utils/autoSlippage.ts

```diff
@@ -0,0 +1,89 @@
+import BigNumber from 'bignumber.js';
+import { assetConstants, chainConstants } from '@/shared/enums';
+import { getDeployedLiquidity, getUndeployedLiquidity } from './pools';
+import { getRequiredBlockConfirmations } from './rpc';
+import { InternalAsset } from '../client';
+
+const getLiquidityAdjustment = async (
+  srcAsset: InternalAsset,
+  destAsset: InternalAsset,
+  amount: bigint,
+) => {
+  const deployedLiquidity = await getDeployedLiquidity(srcAsset, destAsset);
+  const liquidityRatio = new BigNumber(amount.toString()).div(deployedLiquidity.toString());
+
+  if (liquidityRatio.isGreaterThan(0.2)) {
+    return liquidityRatio.multipliedBy(2);
+  }
+  return new BigNumber(0);
+};
+
+const getDepositTimeAdjustment = async (srcAsset: InternalAsset, isBoosted: boolean) => {
+  const { chain } = assetConstants[srcAsset];
+  const { blockTimeSeconds } = chainConstants[chain];
+  const blockConfirmations = isBoosted ? 1 : (await getRequiredBlockConfirmations(srcAsset)) ?? 0;
+  const depositTimeMinutes = (blockConfirmations * blockTimeSeconds) / 60;
+
+  if (['Solana', 'Arbitrum'].includes(chain)) {
+    return depositTimeMinutes * 0.01;
+  }
+  if (['Ethereum', 'Polkadot'].includes(chain)) {
+    return depositTimeMinutes * 0.05;
+  }
+  return depositTimeMinutes * 0.1;
+};
+
+const getUndeployedLiquidityAdjustment = async (asset: InternalAsset, amount: bigint) => {
+  const undeployedLiquidity = await getUndeployedLiquidity(asset);
+  if (undeployedLiquidity >= amount) {
+    return new BigNumber(-0.5);
+  }
+
+  const undeployedRatio = new BigNumber(undeployedLiquidity.toString()).div(amount.toString());
+  if (undeployedRatio.gt(0.5)) {
+    return new BigNumber(-undeployedRatio * 0.1);
+  }
+
+  return new BigNumber(0);
+};
+
+export const calculateRecommendedSlippage = async ({
+  srcAsset,
+  destAsset,
+  intermediateAmount,
+  egressAmount,
+  boostFeeBps,
+  dcaChunks,
+}: {
+  srcAsset: InternalAsset;
+  destAsset: InternalAsset;
+  intermediateAmount?: bigint;
+  egressAmount: bigint;
+  boostFeeBps?: number;
+  dcaChunks: number;
+}) => {
+  const minSlippage = 0.1;
+  const baseSlippage = 1;
+
+  const assetTo = srcAsset === 'Usdc' || destAsset === 'Usdc' ? destAsset : 'Usdc';
```

Comment:
i think i dont understand this yet - does it mean we will only check the liquidity for the first leg?

Reply:
yeah, we only check it for the first leg because it includes the deposit time
we can include the second leg inside the calculations and get the Min (or Max?) value
let's discuss it